src/spec/collapse.directive.spec.ts

```diff
@@ -43,45 +44,75 @@ describe('Directive: Collapse', () => {
     expect(div.classList).toContain('collapse');
   });
 
-  it('should add/remove in class on toggle', () => {
-    expect(element.classList).toContain('in');
-    context.isCollapsed = true;
-    fixture.detectChanges();
-    expect(element.classList).not.toContain('in');
-  });
-
-  it('should be hidden on initialization if isCollapsed = true', () => {
-    context.isCollapsed = true;
+  it('should be hidden on initialization if isCollapsed = true and ' +
+    'should not trigger any animation on initialization', () => {
+    fixture = TestBed.createComponent(TestCollapseComponent);
+    fixture.componentInstance.isCollapsed = true;
     fixture.detectChanges();
+    // animation test by immediate height check
+    element = fixture.nativeElement.querySelector('.collapse');
     expect(element.offsetHeight).toBe(0);
   });
 
-  xit(
-    'should not trigger any animation on initialization if isCollapsed = true',
-    () => {
-      expect(true);
-    }
-  );
-
-  it('should collapse if isCollapsed = true on subsequent use', () => {
+  it('should collapse if isCollapsed = true on subsequent use', done => {
     context.isCollapsed = false;
     fixture.detectChanges();
-    context.isCollapsed = true;
-    fixture.detectChanges();
-    expect(element.offsetHeight).toBe(0);
+    expect(element.offsetHeight).not.toBe(0);
+
+    let subscription: Subscription;
+
+    const detectCollapsed = () => {
+      subscription = context.collapse.collapsed.subscribe(() => {
+        subscription.unsubscribe();
+        expect(element.offsetHeight).toBe(0);
+        done();
+      });
+    };
+    const detectCollapsing = () => {
+      subscription = context.collapse.collapses.subscribe(() => {
+        subscription.unsubscribe();
+        detectCollapsed();
+      });
+    };
+    const triggerCollapse = () => {
+      detectCollapsing();
+      context.isCollapsed = true;
+      fixture.detectChanges();
+    };
+
+    triggerCollapse();
```

Comment:
in tests need to support AAA principle (arrange, act, assert)
I'm not sure why do you need triggerCollapse();

Reply:
Without calling this function the test would not be executed correctly. I grouped different steps  (points in time of the execution) in subfunctions because it seamed easier to read to me and it avoided callback nesting. Due to delay of animations it is required to wait for events and then check the state. One could also extend this test to for non expected events and throw an error in this case. However this arrangement seems to be simpler to read/understand (at least to me).